### PR TITLE
Resolve Task take an exam

### PR DESCRIPTION
**Pull request by mai-123**
#33

### CODE REVIEW

File: app/controllers/tests_controller.rb [hunk 1]
```diff
@@ -1,14 +1,21 @@
 class TestsController < ApplicationController
-  before_action :check_is_logged_in, only: %i[index new create edit]
-  before_action :check_is_admin_permission, only: %i[new create destroy edit]
-  before_action :get_test, only: %i[show destroy edit update]
+  array_method = %i[new create edit update destroy]
+
+  before_action :check_is_logged_in, except: %i[index]
+  before_action :check_is_admin_permission, only: array_method
```
Comment (ducminhv) [inline app/controllers/tests_controller.rb]: can use except ?

File: app/controllers/results_controller.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+class ResultsController < ApplicationController
+  before_action :get_test, :save_result, only: %i[create]
+
+  def create
+    params.require(:questions).each do |answer|
```
Comment (ducminhv) [inline app/controllers/results_controller.rb]: strong params

File: app/controllers/results_controller.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  def save_result_answer(answer, result)
+    @result_answer = ResultAnswer.new
+    @result_answer.answer_id = answer
+    @result_answer.result = result
+    @result_answer.save
+  end
+
+  def update_result(result)
+    count = 0
+    result.result_answers.each do |result_answer|
```
[the remark this answers is not in the document]
Comment (mai-123) [reply app/controllers/results_controller.rb]: use inject return error 
app/controllers/results_controller.rb:32:49: W: Lint/ShadowingOuterLocalVariable: Shadowing outer local variable - count.
    count = result.result_answers.inject(0) do |count, result_answer|
                                                ^^^^^
app/controllers/results_controller.rb:33:7: W: Lint/UselessAssignment: Useless assignment to variable - count.
      count += 1 if result_answer.answer.is_correct?

File: app/controllers/results_controller.rb (hunk 2 above)
Comment (mai-123) [reply app/controllers/results_controller.rb]: it not working

**Pull request merged**